python/sparkdl/graph/builder.py

```diff
@@ -0,0 +1,264 @@
+    :param input_names: names to the input graph elements (must be of Placeholder type)
+    :param output_names: names to the output graph elements
+    """
+
+    def __init__(self, graph_def, input_names, output_names):
+        """
+        :param graph_def: GraphDef object
+        :param input_names: list of input (operation) names (must be typed `Placeholder`)
+        :param output_names: list of output (operation) names
+        """
+        self.graph_def = graph_def
+        self.input_names = input_names
+        self.output_names = output_names
+
+    def dump(self, fpath):
+        """
+        Store the GraphFunction to a file
+
+        :param fpath: str or path, path to the serialized GraphFunction
+        """
+        assert isinstance(fpath, six.string_types)
+
+        gdef_bytes_fpath = "{}.gdef.bytes".format(fpath)
+        with open(gdef_bytes_fpath, 'wb') as fout:
+            gdef_bytes = self.graph_def.SerializeToString()
+            fout.write(gdef_bytes)
+
+        serialized = {"graph_def_file": gdef_bytes_fpath,
+                      "inputs": self.input_names,
+                      "outputs": self.output_names}
+        with open(fpath, 'w') as fout:
+            json.dump(serialized, fout)
+
+    @classmethod
+    def fromSerialized(cls, fpath):
+        """
+        Load an existing GraphFunction from file.
+
+        :param fpath: str or path, path to the serialized GraphFunction
+        """
+        with open(str(fpath)) as fin:
+            serialized = json.load(fin)
+        assert set(['inputs', 'graph_def_file', 'outputs']) <= set(serialized.keys())
+
+        gdef_bytes_fpath = serialized["graph_def_file"]
+        assert os.path.exists(gdef_bytes_fpath), \
+            "TensorFlow GraphDef binary file must be found"
+
+        with open(gdef_bytes_fpath, 'rb') as fin:
+            gdef_bytes = fin.read()
+            gdef = tf.GraphDef.FromString(gdef_bytes)  # pylint: disable=E1101
+
+        gfn = cls(graph_def=gdef,
+                  input_names=serialized["inputs"],
+                  output_names=serialized["outputs"])
+
+        return gfn
+
+    @classmethod
+    def fromKeras(cls, model_or_file_path):
+        """ Build a GraphFunction from a Keras model
+        """
+        def load_model_file(file_path):
+            assert file_path.endswith('.h5'), \
+                'Keras model must be specified as HDF5 file'
+
+            with IsolatedSession(keras=True) as issn:
+                K.set_learning_phase(0) # Testing phase
+                model = load_model(file_path)
+                gfn = issn.asGraphFunction(model.inputs, model.outputs)
+
+            return gfn
+
+        if isinstance(model_or_file_path, KerasModel):
+            model = model_or_file_path
+            try:  # Save to tempdir and restore in a new session
+                model_path = os.path.join(mkdtemp(prefix='kera-'), "model.h5")
```

Comment:
`mkdtemp` should not be inside the try because we do not want to call `rmtree` if `mkdtemp` fails.